### DISCUSSION TITLE
Connect Arduino/Raspberry Pi for GPIO Extention

### DISCUSSION THREAD

**Discussion by josephgiesel** [OpenNeptune3D/OpenNept4une#225] [Q&A]
Hello! I have not done really any mods but I have an Elegoo Neptune 4, and I saw in another discussion that you can connect a Pico or something like that. How can I connect another device for that? Any help is greatly appreciated :)

**Comment by linhungsam** [comment]
I have asked the same question too. It is good to have extensibility on existing hardware through OpenNept4une firmware. Maybe extensibility is already supported by Klipper, but how do we implement it? Pls help us and I can do the testing.

**Reply by josephgiesel** [reply]
I found this from the [Klipper website.](https://www.klipper3d.org/RPi_microcontroller.html#:~:text=If%20you%20want%20to%20use%20the%20host%20as,~%2Fklipper%2F%20sudo%20cp.%2Fscripts%2Fklipper-mcu.service%20%2Fetc%2Fsystemd%2Fsystem%2F%20sudo%20systemctl%20enable%20klipper-mcu.service)
